concrete/src/Entity/File/Version.php

```diff
@@ -2,202 +2,355 @@
+     * The date/time when the file version has been added.
+     *
      * @ORM\Column(type="datetime")
+     *
+     * @var \DateTime
      */
     protected $fvDateAdded;
+
     /**
+     * The date/time when the file version has been approved.
+     *
      * @ORM\Column(type="datetime")
+     *
+     * @var \DateTime
      */
     protected $fvActivateDateTime;
+
     /**
+     * Is this version the approved one for the associated file?
+     *
      * @ORM\Column(type="boolean")
+     *
+     * @var bool
      */
     protected $fvIsApproved = false;
+
     /**
+     * The ID of the user that created the file version.
+     *
      * @ORM\Column(type="integer")
+     *
+     * @var int
      */
     protected $fvAuthorUID = 0;
+
     /**
-     * @ORM\Column(type="bigint")
-     */
-    protected $fvSize = 0;
-    /**
+     * The ID of the user that approved the file version.
+     *
      * @ORM\Column(type="integer")
+     *
+     * @var int
      */
     protected $fvApproverUID = 0;
+
+    /**
+     * The size (in bytes) of the file version.
+     *
+     * @ORM\Column(type="bigint")
+     *
+     * @var int
+     */
+    protected $fvSize = 0;
+
     /**
+     * The title of the file version.
+     *
      * @ORM\Column(type="string", nullable=true)
+     *
+     * @var string|null
      */
     protected $fvTitle = null;
+
     /**
+     * The description of the file version.
+     *
      * @ORM\Column(type="text", nullable=true)
+     *
+     * @var string|null
      */
     protected $fvDescription = null;
+
     /**
+     * The extension of the file version.
+     *
      * @ORM\Column(type="string", nullable=true)
+     *
+     * @var string|null
      */
     protected $fvExtension = null;
 
     /**
-     * @ORM\Column(type="integer")
-     */
-    protected $fvType = 0;
-    /**
+     * The tags assigned to the file version (separated by a newline character - '\n').
+     *
      * @ORM\Column(type="text", nullable=true)
+     *
+     * @var string|null
      */
     protected $fvTags = null;
+
+    /**
+     * The type of the file version.
+     *
+     * @ORM\Column(type="integer")
+     *
+     * @var int
+     */
+    protected $fvType = 0;
+
     /**
+     * Does this file version has a thumbnail to be used for file listing?
+     *
      * @ORM\Column(type="boolean")
+     *
+     * @var bool
      */
     protected $fvHasListingThumbnail = false;
+
     /**
+     * Does this file version has a thumbnail to be used used for details?
+     *
      * @ORM\Column(type="boolean")
+     *
+     * @var bool
      */
     protected $fvHasDetailThumbnail = false;
 
+    /**
+     * The currently loaded Image instance.
+     *
+     * @var \Imagine\Image\ImageInterface|false|null null: still not loaded; false: load failed; ImageInterface otherwise
+     */
     private $imagineImage = null;
 
+    /**
+     * Initialize the instance.
+     */
+    public function __construct()
+    {
+        $this->fvDateAdded = new DateTime();
```

Comment:
Should this be a DateTimeImmutable?

Reply:
I don't know, this would be a breaking change